packages/grafana-data/src/events/common.ts

```diff
@@ -0,0 +1,33 @@
+import { DataFrame } from '../types';
+import { eventFactory } from './eventFactory';
+import { BusEventWithPayload } from './types';
+
+/** @alpha */
+export interface DataHoverPayload {
+  raw: any; // Original mouse event (includes pageX etc)
+
+  x: Record<string, number>; // { time: 5678 },
+  y: Record<string, number>; // { __fixed: 123, lengthft: 456 }  // each axis|scale gets a value
+
+  data?: DataFrame; // source data
+  rowIndex?: number; // the hover row
+  columnIndex?: number; // the hover column
+
+  // Drag events
+  down?: Omit<DataHoverPayload, 'down'>;
```

Comment:
not sure what your thinking here, a nested event for where we started?

Reply:
yes -- this would be the "drag" event.  where the mouse was when the event started